### DISCUSSION TITLE
Should JSON loader preserve the __proto__ property?

### DISCUSSION THREAD

**Discussion by hai-x** [webpack/webpack#19100] [Ideas]
Webpack native support JSON file. When import JSON file, we treat it as js object and parsed by JSON.parse. But `__proto__`  behaves differently in [`object_literal_syntax_vs` and `.json`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#object_literal_syntax_vs._json).

> In JSON, "__proto__" is a normal property key. In an object literal, it [sets the object's prototype](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#prototype_setter).

So my idea is If we treat it as a js object, should we handle `__proto__` key specially?

Related: https://github.com/evanw/esbuild/issues/3700

**Comment by hai-x** [comment]
@alexander-akait A friendly ping, what do you think?

**Reply by alexander-akait** [reply]
@hai-x Sorry for delay

> So my idea is If we treat it as a js object, should we handle __proto__ key specially?

I think no, why do you think differently?

**Reply by hai-x** [reply]
> why do you think differently?

In JSON, "proto" is a normal property key. In an object literal, it [sets the object's prototype](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#prototype_setter).

**Reply by hai-x** [reply]
> I think no

gotcha.

**Reply by alexander-akait** [reply]
I think we have a test for such things, need verify :smile:

**Comment by hai-x** [comment]
Completed in https://github.com/webpack/webpack/pull/19232.
